pkg/controller/registry/resolver/cache/cache.go

```diff
@@ -52,6 +52,11 @@ type SourceProvider interface {
 	Sources(namespaces ...string) map[SourceKey]Source
 }
 
+type SourceProviderWithInvalidate interface {
+	SourceProvider
+	Invalidate(key SourceKey)
+}
```

Comment:
Since Invalidate isn't used by the cache package, can we do without declaring this interface type here?

It might be reasonable to define an interface beside `OperatorGroupToggleSourceProvider` that looks like

```go
type SourceInvalidator interface {
 Invalidate(SourceKey)
}
```

then change `OperatorGroupToggleSourceProvider` to accept plain old `cache.SourceProvider` as its wrapped type. Then, in `(OperatorGroupToggleSourceProvider).Invalidate(SourceKey)`, do a type-assertion on the wrapped value to see if it implements `SourceInvalidator` to decide whether or not to delegate the invalidate call.

This would be slightly more effort, but idiomatic. More importantly, it avoids expanding the Go API surface of resolver and its subpackages. There's still interest in treating those as a shared library.

Reply:
Could we just move it to operator.go and define it as a client side interface?